go-controller/pkg/kube/kube.go

```diff
@@ -374,6 +374,13 @@ func (k *KubeOVN) GetEgressIPs() (*egressipv1.EgressIPList, error) {
 	})
 }
 
+// GetCloudPrivateIPConfigs returns the list of all CloudPrivateIPConfig objects from kubernetes
+func (k *KubeOVN) GetCloudPrivateIPConfigs() (*ocpcloudnetworkapi.CloudPrivateIPConfigList, error) {
+	return k.CloudNetworkClient.CloudV1().CloudPrivateIPConfigs().List(context.TODO(), metav1.ListOptions{
```

Comment:
we need to fetch from kapi because informers for cloudPrivateIP have not started? hope this list call doesn't cost us a performance at startup

Reply:
yes, i guess this retrieval is done via kube api, but it happens for every egress ip event. this is what more concerning.

Reply:
wait not sure I follow... the retrieval is done for every event? I thought this is called only once on startup right? Isn't this whole logic a one time cleanup Stale stuff on master restarts a.k.a start ups only?

Reply:
yes, yes. this is being invoked only once at the watcher startup, so we are safe here.